docker-app/qfieldcloud/core/utils2/storage.py

```diff
@@ -96,3 +100,54 @@ def remove_project_thumbail(project: "Project") -> None:  # noqa: F821
     bucket = qfieldcloud.core.utils.get_s3_bucket()
     key = project.thumbnail_uri
     bucket.object_versions.filter(Prefix=key).delete()
+
+
+def purge_old_file_versions(project: "Project") -> None:  # noqa: F821
+    """
+    Deletes old versions of all files in the given project. Will keep __3__
+    versions for COMMUNITY user accounts, and __10__ versions for PRO user
+    accounts
+    """
+
+    logger.info(f"Cleaning up old files for {project}")
+
+    # Determine account type
+    account_type = project.owner.useraccount.account_type
+    if account_type == UserAccount.TYPE_COMMUNITY:
+        keep_count = 3
+    elif account_type == UserAccount.TYPE_PRO:
+        keep_count = 10
+    else:
+        raise NotImplementedError(f"Unknown account type {account_type}")
+
+    logger.debug(f"Keeping {keep_count} versions")
+
+    # Process file by file
+    for file in qfieldcloud.core.utils.get_project_files_with_versions(project.pk):
+
+        filename = file.latest.name
+        old_versions = file.versions
```

Comment:
Can you use `file.versions`  below directly, it's easier to keep track where it comes from, especially with typing enabled.